IFC 4.3/ProjectSetup_1/UT_ProjectSetup_1.ifc

```diff
@@ -0,0 +1,48 @@
+ISO-10303-21;
+
+/* ------- This file was produced manually for IFC Infra Extension Deployment project. --------- */
+/* ------- The file should be come with a README.md with supplementing info. ------------------- */
+/* ------- Stefan Jaud, stefan.jaud@tum.de, 2020.07.15. ---------------------------------------- */
+
+HEADER;
+FILE_DESCRIPTION (('ViewDefinition [NotDefinedYet]'), '2;1');
+FILE_NAME ('UT_ProjectSetup_1.ifc', '2020-07-15T18:00:00', (''), (''), 'Stefan Jaud manually', 'SublimeText 3.0', '');
+FILE_SCHEMA (('IFC4X3_RC1'));
+ENDSEC;
+
+DATA;
+
+/* ------- General definitions ----------------------------------------------------------------- */
+#1 = IFCDIMENSIONALEXPONENTS(0, 0, 0, 0, 0, 0, 0);
+/* ------- Global X direction ------------------------------------------------------------------ */
+#2 = IFCDIRECTION((1., 0., 0.));
+/* ------- Global Y direction ------------------------------------------------------------------ */
+#3 = IFCDIRECTION((0., 1., 0.));
+/* ------- Global Z direction ------------------------------------------------------------------ */
+#4 = IFCDIRECTION((0., 0., 1.));
+/* ------- Global origin ----------------------------------------------------------------------- */
+#5 = IFCCARTESIANPOINT((0., 0., 0.));
+
+/* ------- IfcProject is the top most element in the project structure ------------------------- */
+#16 = IFCPROJECT('2DAvEupIz0HQr73cMaawtY', $, 'BoilerPlate', 'The bare minimum of an IFC file exchange with no content, only boilerplate code.', $, $, $, (#28), #17);
+
+/* ------- IfcUnitAssignment defines the global units for measures and values ------------------ */
+/* ------- when the units are not otherwise defined -------------------------------------------- */
+#17 = IFCUNITASSIGNMENT((#18, #19, #20, #24, #25, #26, #27));
+#18 = IFCSIUNIT(*, .LENGTHUNIT., $, .METRE.);
+#19 = IFCSIUNIT(*, .AREAUNIT., $, .SQUARE_METRE.);
+#20 = IFCSIUNIT(*, .VOLUMEUNIT., $, .CUBIC_METRE.);
+#21 = IFCDIMENSIONALEXPONENTS(0, 0, 0, 0, 0, 0, 0);
+#22 = IFCSIUNIT(*, .PLANEANGLEUNIT., $, .RADIAN.);
+#23 = IFCMEASUREWITHUNIT(IFCPLANEANGLEMEASURE(1.5707963267949E-2), #22);
```

Comment:
should be:
#23= IFCMEASUREWITHUNIT(IFCPLANEANGLEMEASURE(0.017453293),#22);

conversion rate between degree and radians. Needed, if all angle measures within the file are provided as degree (0 .. 360).

Reply:
Minor nitpick. This is `pi / 180` but please use full ieee 754 double precision for this as you might get into tolerance issues otherwise: `0.017453292519943295`

Reply:
Isn't it a bit confusing that the base unit is radian and a 'degree' conversion is being applied to it to get gradian?

Reply:
next line says:
#24 = IFCCONVERSIONBASEDUNIT(#21, .PLANEANGLEUNIT., 'degree', #23);

if it is using gradian (or gon), then this line needs to be corrected into:
#24 = IFCCONVERSIONBASEDUNIT(#21, .PLANEANGLEUNIT., 'gradian', #23);

NOTE: it is a potential danger, as most software has implemented degree, and I am unsure, whether this string value is really checked. Would need a good explanation.

Reply:
in addition, if needed, an agreement has to be documented on its name "gradient" or "gon" and it should be documented here: https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC1/HTML/link/ifcconversionbasedunit.htm

Reply:
From my experience, this value is usually capitalized. However, it should be case sensitive. It might be the case that nobody checks it.

It would make sense to have an agreement if this value will be used.

Reply:
Nice catch! It was gon -> radian, not degree -> radian. I've changed the conversion factor to correspond to "degree" in both examples. Resolved in f9b5d5d.

Reply:
@TLiebich Care to put your suggestion forward as an issue in NextGen / some other repo?